Dockerfile.lab

```diff
@@ -12,30 +12,26 @@ RUN apt-get -qq update && apt-get install -y --no-install-recommends \
     vim-tiny \
     lsof && \
   rm -rf /var/lib/apt/lists/*
+
 USER $NB_USER
 
+RUN conda create -n jupyterlab-ext cookiecutter git -y -c conda-forge && \
+  echo "Activate conda envrironment jupyterlab-ext" && \
+  /bin/bash -c "source activate jupyterlab-ext" && \
+  echo "Install nodejs 14" && \
+  mamba install -c conda-forge -y nodejs=14 && \ 
+  echo "Install jupyterlab 3.6.3" && \
+  mamba install -c conda-forge -y jupyterlab=3.6.3
```

Comment:
why are we installing jupyterlab here and not starting from singleuser:3.6.3? Does this result in two different version of jupyterlab being installed?

Reply:
problem with docker images is that if you install new software it makes a new layer, so the old version still exists and gets pulled, so the image that is pulled eventually will have both versions of jupyterlab installed.

Reply:
Do we need the latest version of jupyter-leaflet? can we maybe use an older version that worked in the past, and just pin it to that version for now?

Reply:
The problem is that singleuser/jupyterhub image doesn't contain jupyterlab-manager and jupyter-leaflet. So when I tried to installe those extensions directly to the base image (jupyterhub/singleuser) there were errors when I deployed to jupyterhub in incore, even though it worked well just running the incore-lab image built on jupyterhub/singleuser image just installing the extensions on somaek. This eventually fixed by installing jupyterlab inside the jupyterhub/singleuser image, instead of installing jupyterlab-manger and jupyter-leaflet in the jupyterhub/singleuser image.

Reply:
The problem is that if I just use the existing jupyterhub/singleuser and install the jupyter-manager and jupyter-leaflet in it, it only works when it runs as a jupyterlab itself. When I put the image inside jupyterhub, it somehow fails to activate those extensions. However, it works without any problem when I installs the jupyterlab in the jupyterhub/singleuser image that is not very desirable maybe. I have tested several versions of manager and leaftlet and most of them worked well when it ran as jupyterlab from somaek that was outside of the jupyterhub. However, when I deployed the same image to the incore, that is under jupyterhub, it failed again. Only successful instance was when I installed jupyterlab insided the singleuser image. I don't know why... but it was the only working way that I have found so far.